### ISSUE TITLE
Yuzu games become choppy after switching to/from external monitor

### ISSUE THREAD

**Issue by Hexaae** [yuzu-emu/yuzu#13023]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

all

### Description of Issue

Starting from this old bug #8639 some time ago with Zelda: Link's Awakening...
... I've been able to reproduce the issue again even with modern drivers (current 551.23 SD) and recent Yuzu mainline builds, and I've found the real cause of this choppiness issue.
The issue happens if you have an external monitor attached (I use a gaming laptop with GTX 1070 8GB Max-P, internal screen 1920x1080@144Hz G-Sync + miniDP-1.4 to DP connected to an external monitor 3440x1440@100Hz Freesync) and you switch to external monitor at least once, and then turn off external monitor to return to laptop display.
At this point, yuzu performance are doomed. Notice this is independent from yuzu running or not in the process (i.e. you can quit yuzu before you switch to external monitor and relaunch it only once you're back to the laptop internal monitor), so it must be something between Yuzu VS NVidia driver status when switching screens. Doesn't happen with any other emu like CEMU or with other Vulkan/DX12/OpenGL games.

GPU core/memory clocks didn't change before/after when playing with yuzu so it's not a stuck GPU clock issue.
Tested different NVidia drivers new and old and it's always the same, even with 531.61...
The only solution to make yuzu run fine again once I switched to the external monitor, is to reboot... or disable/enable the GTX1070 from Device Manager (!!) and then yuzu will immediately run fine again, at least 'till I switch again to the external monitor. That's why I think is some strange bug between yuzu VS NVidia drivers. _It's like something has changed in the video drivers status after switching monitor, and this affects yuzu (only!) with degraded performance and hiccups._

### Expected Behavior

No impact on yuzu switching to/from external monitor or not.

### Reproduction Steps

REPRO STEPS:
- reboot your PC
- switch to external monitor (turn it on and the laptop screen will go black moving your Win desktop to the external monitor)
- switch back to laptop screen (i.e. turning off external monitor...)
- run Yuzu for the first time since reboot, with any game, and see degraded performance with choppy cyclical "hiccups" when playing any game, e.g. Metroid Dread, Zelda titles, Kirby titles etc.

### Log File

[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/14302495/yuzu_log.txt)

### System Configuration

CPU: i7-8750h
GPU/Driver: 551.23 SD
RAM: 32GB
OS: Win11 23H2

**Comment by liamwhite** [comment]
If you rename yuzu.exe to Cemu.exe does anything change

**Comment by Squall-Leonhart** [comment]
This happens when the refresh rate changes during the process of changing displays.

**Comment by Hexaae** [comment]
But why is yuzu affected, even if not running?
When I launch yuzu and is running I don't change refresh rate. I can reproduce the problem switching screen BEFORE I run yuzu...

**Comment by Hexaae** [comment]
In the meanwhile I did another test: set external monitor 21:9 to 144Hz (like internal G-Sync display), but the bug with yuzu choppy performance was still reproducible with both screens at 144Hz (and yuzu launched AFTER screen switch back to internal display).

**Comment by liamwhite** [comment]
Does it have the same problem with g-sync completely off everywhere?

**Comment by Hexaae** [comment]
> Does it have the same problem with g-sync completely off everywhere?

Yes, just tested with G-Sync off and still happens after I switch 16:9 144Hz -> 21:9 100Hz -> 16:9 144Hz

**Comment by Hexaae** [comment]
Just for the chronicles, tested in the meanwhile also with ryujinx and no issue.

**Comment by Hexaae** [comment]
Yes, tried that too :(

**Comment by Hexaae** [comment]
Can you help me launch a game with yuzu-cmd using the same config + updates + mods I use from the UI edition?
-config option is vague and I don't see options to load also updates etc...
If I run yuzu-cmd.exe -f -g "roms\game.nsp" it will open a black screen with music only, or trying another game won't pass the intro screen... (??)

**Comment by liamwhite** [comment]
It might be having issues due to rendering in OpenGL, you'll have to edit the key in sdl2-config.ini Renderer.backend=1. The updates should be applied I think, but I don't think mods get loaded.

**Comment by Hexaae** [comment]
Can't find that key in my [%AppData%\yuzu\config\sdl2-config.ini ](https://paste.sh/sXYrLc2K#JziFwzXw0SYMSRChxTBU9YR_)

**Comment by liamwhite** [comment]
backend=1 is vulkan, so that's fine. But vulkan_device=0 might be the iGPU.

**Comment by Hexaae** [comment]
did some more tests and I can say yuzu-cmd.exe definitely runs just like yuzu.exe before the screen changes bug... so it could be a problem in the UI code when initializing the screen (in yuzu.exe)?
